presto-hive/src/main/java/io/prestosql/plugin/hive/GenericHiveRecordCursor.java

```diff
@@ -383,7 +388,7 @@ else if (fieldValue instanceof HiveCharWritable) {
                 hiveValue = ((HiveCharWritable) fieldValue).getTextValue();
             }
             else {
-                throw new IllegalStateException("unsupported string field type: " + fieldValue.getClass().getName());
```

Comment:
This should also be a programming error of some sort and should never occur. If you've ever seen this occur, we want to know about it.